src/components/VmDetail/index.js

```diff
@@ -8,9 +8,36 @@ import { closeVmDetail } from '../../actions/vm'
 
 import VmIcon from '../VmIcon'
 import VmDisks from '../VmDisks'
-
+import Time from '../Time'
 import VmActions from '../VmActions'
 
+const LastMessage = ({ vmId, userMessages }) => {
+  const vmMessages = userMessages.get('records')
+    .filter(msg => { return (msg.failedAction && msg.failedAction.payload && msg.failedAction.payload.vmId === vmId) })
```

Comment:
please make lambdas returning single value without the braces and `return`. Just `msg => expression` is sufficient.